consensus/src/sync/live/state_queue/mod.rs

```diff
@@ -523,6 +523,33 @@ impl<N: Network> Stream for StateQueue<N> {
         loop {
             match self.chunk_request_component.poll_next_unpin(cx) {
                 Poll::Ready(Some((chunk, start_key, peer_id))) => {
+                    if start_key.len() >= 8 {
+                        // Extract the first 4 bytes (8 nibles ) of the key nibbles
+                        let key =
+                            i32::from_str_radix(&start_key.slice(0, 8).to_string(), 16).unwrap();
```

Comment:
Can probably replace the whole `if-else` thing with `let key = u32::from_str_radix(&format!("{}00000000", start_key)[..8].unwrap()`

Reply:
I really like this!
Just made a small change to take into consideration when the start key is "ε"